cpp/include/cudf/utilities/error.hpp

```diff
@@ -114,10 +114,10 @@ namespace detail {
 
 inline void throw_cuda_error(cudaError_t error, const char* file, unsigned int line)
 {
-  // Calls cudaGetLastError twice. It is nearly certain that a fatal error occurred if the second
-  // call doesn't return with cudaSuccess.
+  // Calls cudaGetLastError to clear the error status. It is nearly certain that a fatal error
+  // occurred if it still returns the same error after a cleanup.
   cudaGetLastError();
-  auto const last = cudaGetLastError();
+  auto const last = cudaFree(0);
```

Comment:
Does this end up doing a full device synchronize as normal `cudaFree` calls do?

Reply:
Hi @jlowe, according to the CUDA doc, "If devPtr is 0, no operation is performed. [cudaFree()](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1ga042655cbbf3408f01061652a075e094) returns cudaErrorValue in case of failure."

Reply:
If we're guaranteed this doesn't do anything slow like a synchronize it seems OK to me, but I'll defer to @jrhemstad's judgement on whether this is the best approach with the limited tools we have to detect this.

Reply:
Shall we just let the PR in, as a sort of workaround?